### PR TITLE
Fix connection pools not shared between writer -> replica during tests:

### PR DESCRIPTION
**Pull request by Edouard-chin**
Fix connection pools not shared between writer -> replica during tests:

- ### Problem

  Connection pools are not properly shared. A replica can't see the
  data in the open transaction on the writing connection.
  ```ruby
  Dog.create!(name: 'bilou')

  AnimalsBase.connected_to(role: :reading) do
    Dog.find_by(name: 'bilou') # No result
  end
  ```

  The reason of this bug is because when test starts, we iterate
  over `AR::Base.connection_handlers` which only contains the `writing`
  handler since models haven't been loaded (app isn't eagerloaded
  test env).

  ### Solution

  Share connection pools as soon as a connection is established
  by creating a subcriber.

cc/ @rafaelfranca @casperisfine @eileencodes (FYI issue you saw in https://github.com/rails/rails/issues/37765#issuecomment-557253443 was related to this)

### CODE REVIEW

File: activerecord/test/cases/fixtures_test.rb [hunk 1]
```diff
@@ -1399,13 +1399,6 @@ class MultipleDatabaseFixturesTest < ActiveRecord::TestCase
       rw_conn = ActiveRecord::Base.connection
       ro_conn = ActiveRecord::Base.connection_handlers[:reading].connection_pool_list.first.connection
 
-      assert_not_equal rw_conn, ro_conn
-
-      enlist_fixture_connections
-
-      rw_conn = ActiveRecord::Base.connection
-      ro_conn = ActiveRecord::Base.connection_handlers[:reading].connection_pool_list.first.connection
```
Comment (eileencodes) [inline activerecord/test/cases/fixtures_test.rb]: Why did you delete the test that ensures this works? If this test isn't correct can you create a test that is more correct then?

**Review by eileencodes — CHANGES_REQUESTED** [review]

File: activerecord/test/cases/fixtures_test.rb (hunk 1 above)
Comment (Edouard-chin) [reply activerecord/test/cases/fixtures_test.rb]: This part of the test is no longer necessary. The test is still checking if both pools are the same between replica->master https://github.com/rails/rails/pull/37774/files#diff-9a795b6442abbee1bf369da52a59961cR1402

What I deleted is the check to make sure both pool aren't the same which is now useless.

**Review by eileencodes — APPROVED** [review]

**Comment by eileencodes** [comment]
Backported to 6-0-stable since this fixes a bug.
